tcms/rpc/tests/test_testexecution.py

```diff
@@ -274,6 +276,77 @@ def test_get_links(self):
         self.assertEqual(logs[0]['url'], 'http://kiwitcms.org')
 
 
+class TestExecutionHistory(APITestCase):
+    def _fixture_setup(self):
+        super()._fixture_setup()
+
+        self.execution = TestExecutionFactory()
+
+    def test_history_for_non_existing_execution(self):
+        with self.assertRaisesRegex(XmlRPCFault, "TestExecution matching query does not exist."):
+            self.rpc_client.TestExecution.history(-5)
+
+    def test_history_new_execution(self):
+        execution = TestExecutionFactory()
+
+        history = self.rpc_client.TestExecution.history(execution.pk)
+
+        self.assertEqual(1, len(history))
+
+    def test_history(self):
+        """
+        Test that for an execution that has been updated 3 times,
+        there are 4 history entries (first one is the creation of the object).
+
+        Note: the `time.sleep` call after every update is necessary,
+        because otherwise the changes happen too fast,
+        and the rpc serializator returns the dates as 'xmlrpc.client.DateTime'
```

Comment:
:+1: for the explanation, just that it's not the rpc serializer (in terms of which library we use or even how we export the data). It is the XML-RPC protocol and its adherence to ISO standards for representing date-time values. See my previous comment.

Reply:
got it. is the explanation better now?